dev/io.openliberty.microprofile.reactive.messaging.internal_fat/fat/src/io/openliberty/microprofile/reactive/messaging/fat/kafka/metrics/MetricsTest.java

```diff
@@ -0,0 +1,151 @@
+        server.startServer();
+    }
+
+    @Test
+    public void testDeliverPayload() throws Exception {
+        // Test that metrics are updated when a payload is delivered via an emitter
+        runTest(server, APP_NAME + "/MetricsTest", "emitterDeliverPayload");
+        HashMap<String, Integer> metrics = getMetrics();
+
+        assertThat("5 messages not detected on channel", metrics.get("metrics-incoming") == 5);
+//      Assertion disabled as emitters currently don't report metrics per spec
+//        assertTrue("Outgoing and incoming channel counts are not the same",
+//                   metrics.get("mp_messaging_message_count_total{channel=\"MetricsEmitter\",mp_scope=\"base\",}") == metrics
+//                                   .get("mp_messaging_message_count_total{channel=\"metrics-incoming\",mp_scope=\"base\",}"));
+    }
+
+    @Test
+    public void testDeliverMessage() throws Exception {
+        // Test that metrics are updated when a message is delivered via delivery and reception beans
+        runTest(server, APP_NAME + "/MetricsTest", "deliverMessage");
+        HashMap<String, Integer> metrics = getMetrics();
+
+        assertThat("5 messages not detected on channel", metrics.get("metrics-outgoing") == 5);
+        assertThat("Outgoing and incoming channel counts are not the same",
+                   metrics.get("metrics-outgoing") == metrics
+                                   .get("metrics-message-incoming"));
+    }
+
+    @AfterClass
+    public static void teardown() throws Exception {
+        try {
+            server.stopServer();
+        } finally {
+            KafkaUtils.deleteKafkaTopics(KafkaTests.getAdminClient());
+        }
+    }
+
+    private HashMap<String, Integer> getMetrics() throws Exception {
+        // Returns the MicroProfile Reactive Messaging metrics from the /metrics endpoint
+        HttpURLConnection response = HttpUtils.getHttpConnection(new URL("http", server.getHostname(), 8010, "/metrics"),
+                                                                 200, null, 30, HTTPRequestMethod.GET, null, null);
+        HashMap<String, Integer> metrics = new HashMap<String, Integer>();
+        response.setReadTimeout(30 * 1000);
+        InputStream result = response.getInputStream();
+        BufferedReader in = new BufferedReader(new InputStreamReader(result));
+        String currentLine;
+
+        // filter the /metrics output for reactive messaging metrics
+        while ((currentLine = in.readLine()) != null)
+            if (currentLine.contains("mp_messaging_message_count_total{channel=")) {
+                String channelString = currentLine.split("channel=\"")[1].replace(",mp_scope=\"base\",", "");
+                // the metric counter is always a whole number (int)
+                // extract channel name and counter total from response line
+                metrics.put(channelString.split("\"} ")[0], (int) Float.parseFloat(channelString.split("\"} ")[1]));
+            }
```

Comment:
This might be easier with a regex and capturing groups.

```java
Matcher m = Pattern.compile("mp_messaging_message_count_total{.*channel=\"([^\"])\".*} +(.*)").matcher(currentLine);
if (m.find()) {
    metrics.put(m.group(1), (int) Float.parseFloat(m.group(2)));
}
```

Reply:
I had a play around with this but I can't seem to get it working with regex when `,mp_scope="base\,"` is present.